staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -32,6 +33,58 @@ import (
 	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+var (
+	// Private copy of jsoniter.ConfigCompatibleWithStandardLibrary to try to shield against possible mutations
+	// from outside. Still does not protect from package level jsoniter.Register*() functions - someone calling them
+	// in some other library will mess with every usage of the jsoniter library in the whole program.
+	// See https://github.com/json-iterator/go/issues/265
+	configCompatibleWithStandardLibrary = jsoniter.Config{
+		EscapeHTML:             true,
+		SortMapKeys:            true,
+		ValidateJsonRawMessage: true,
+	}.Froze()
+)
+
+func init() {
+	// Force jsoniter to decode number to interface{} via int64/float64, if possible.
+	configCompatibleWithStandardLibrary.RegisterExtension(&customNumberExtension{})
+}
+
+type customNumberExtension struct {
+	jsoniter.DummyExtension
+}
+
+func (cne *customNumberExtension) CreateDecoder(typ reflect2.Type) jsoniter.ValDecoder {
+	if typ.String() == "interface {}" {
+		return customNumberDecoder{}
+	}
+	return nil
+}
+
+type customNumberDecoder struct {
+}
+
+func (customNumberDecoder) Decode(ptr unsafe.Pointer, iter *jsoniter.Iterator) {
+	switch iter.WhatIsNext() {
+	case jsoniter.NumberValue:
+		var number jsoniter.Number
+		iter.ReadVal(&number)
+		i64, err := strconv.ParseInt(string(number), 10, 64)
+		if err == nil {
+			*(*interface{})(ptr) = i64
+			return
+		}
+		f64, err := strconv.ParseFloat(string(number), 64)
+		if err == nil {
+			*(*interface{})(ptr) = f64
+			return
+		}
+		iter.ReportError("DecodeNumber", err.Error())
```

Comment:
This line is missing in the deleted snippet below. Everything else is exactly the same.